### PR TITLE
Separated out external and internal interfaces for Transporter and Dispatcher

### PR DESCRIPTION
**Pull request by jzila**

### CODE REVIEW

File: rpc2/transport.go [hunk 1]
```diff
@@ -13,58 +13,64 @@ type WrapErrorFunc func(error) interface{}
...
-	dispatcher Dispatcher
+	dispatcher dispatcher
 	packetizer *Packetizer
 	log        LogInterface
 	running    bool
 	wrapError  WrapErrorFunc
 }
 
-func NewConPackage(c net.Conn, mh *codec.MsgpackHandle) *ConPackage {
+func NewconPackage(c net.Conn, mh *codec.MsgpackHandle) *conPackage {
```
Comment (akalin-keybase) [inline rpc2/transport.go]: I think the usual pattern is `newConPackage`?

File: rpc2/transport.go [hunk 2]
```diff
@@ -204,27 +210,25 @@ func (t *Transport) ReadByte() (b byte, err error) {
...
 	}
 	return err
 }
 
-func (t *Transport) GetDispatcher() (d Dispatcher, err error) {
-	t.run(true)
+func (t *Transport) getDispatcher() (dispatcher, error) {
 	if !t.IsConnected() {
-		err = DisconnectedError{}
+		return nil, DisconnectedError{}
```
Comment (akalin-keybase) [inline rpc2/transport.go]: I think govet or golint will complain about this, and favors:

```
if ... {
  return ...
}
return ...
```

**Comment by akalin-keybase** [comment]
Went over this to familiarize myself with the code, found a couple of nits.

File: rpc2/transport.go (hunk 1 above)
Comment (akalin-keybase) [inline rpc2/transport.go]: Never mind, just saw your newer pull request changes this!

File: rpc2/transport.go (hunk 2 above)
Comment (jzila) [inline rpc2/transport.go]: Fair, I'll change it.

Comment (maxtaco) [inline rpc2/transport.go]: @akalin-keybase brings up a good point, lintable code is also good wherever possible.  Again, not something I was wise enough to do at first.

**Pull request merged**